src/controllers/customerController.js

```diff
@@ -1,67 +1,140 @@
-/**
- * uncomment this after schema designed compeletly 
- * and pulled intpo the repository and delete this comment
- */
-// const customerSchema = require('../models/customerSchema')
+const customerSchema = require('../models/customerSchema');
 
 /**
- * This return all Customers 
+ * This returns all Customers
  * @param {*} req 
  * @param {*} res 
  */
-async function getAllCustomers (req, res) {
-
-    //implemant the code here
-
+async function getAllCustomers(req, res) {
+  try {
+    const customers = await customerSchema.find();
+    res.status(200).json(customers);
+  } catch (error) {
+    res.status(500).json({ error: error.message });
+  }
 }
 
 /**
- * Add new Customer into the Customer schema
+ * Add a new Customer to the Customer schema
  * @param {*} req 
  * @param {*} res 
  */
-async function addCustomer (req, res) {
-
-    //implemant the code here
+async function addCustomer(req, res) {
+  try {
+    // Extract customer data from the request body
+    const { customer_email, customer_name, customer_address, birthdate, provider, provider_id } = req.body;
```

Comment:
Please delete provider and provider_id here because those are not needed at this stage of the project.  
Could add the password to the Schema? And change the code here based on that?